lib/Parser/Parse.cpp

```diff
@@ -6033,13 +6035,22 @@ void Parser::UpdateOrCheckForDuplicateInFormals(IdentPtr pid, SList<IdentPtr> *f
 }
 
 template<bool buildAST>
-void Parser::ParseFncFormals(ParseNodePtr pnodeFnc, ushort flags)
+void Parser::ParseFncFormals(ParseNodePtr pnodeFnc, ParseNodePtr pnodeParentFnc, ushort flags)
```

Comment:
Does this code work if we're parsing an arrow function contained in some number of other arrow functions? Does looking at the immediate parent tell us what we need to know?